samples: openthread: enable rebooting to bootloader for nRF52 Dongle (#13965)
nrfconnect/sdk-nrf: samples/openthread/coprocessor/README.rst

The Co-processor sample enables a diagnostic module in a similar way as described in the :ref:`ot_cli_sample_diag_module` section of the :ref:`ot_cli_sample` sample documentation.
However, the Co-processor and CLI samples use different commands for the module, as described in the :ref:`ot_coprocessor_testing` section.

Rebooting to bootloader
=======================

> The Co-processor sample enables rebooting to bootloader for the ``nrf52840dongle_nrf52840`` build target, in a similar way as described in the :ref:`ot_cli_sample_bootloader` section of the :ref:`ot_cli_sample` sample documentation.

Reviewer: ```suggestion
The Co-processor sample enables rebooting to bootloader for the ``nrf52840dongle_nrf52840`` build target, similar to what is described in the :ref:`ot_cli_sample_bootloader` section of the :ref:`ot_cli_sample` sample documentation.
```

However, the Co-processor and CLI samples use different commands, as described in the :ref:`ot_coprocessor_testing` section.

> Additionally, posix application described in :ref:`ug_thread_tools_ot_apps` should be built with ``-DOT_PLATFORM_BOOTLOADER_MODE=ON`` option.

Reviewer: ```suggestion
Additionally, the :ref:`ug_thread_tools_ot_apps` should be built with ``-DOT_PLATFORM_BOOTLOADER_MODE=ON`` option.
```
